data-protocols/dsp/dsp-negotiation/dsp-negotiation-api/src/main/java/org/eclipse/edc/protocol/dsp/negotiation/api/controller/DspNegotiationApiController.java

```diff
@@ -107,40 +112,53 @@ public DspNegotiationApiController(String callbackAddress,
      *
      * @param id    of contract negotiation.
      * @param token identity token.
+     * @return the requested contract negotiation or an error.
      */
     @GET
     @Path("{id}")
-    public Map<String, Object> getNegotiation(@PathParam("id") String id, @HeaderParam(AUTHORIZATION) String token) {
-        checkAndReturnAuthToken(token);
-        throw new UnsupportedOperationException("Currently not supported.");
+    public Response getNegotiation(@PathParam("id") String id, @HeaderParam(AUTHORIZATION) String token) {
+        return errorResponse(Optional.of(id), Response.Status.NOT_IMPLEMENTED, NOT_IMPLEMENTED);
     }
 
     /**
      * Provider-specific endpoint.
      *
      * @param jsonObject dspace:ContractRequestMessage sent by a consumer.
      * @param token      identity token.
+     * @return the created contract negotiation or an error.
      */
     @POST
     @Path(INITIAL_CONTRACT_REQUEST)
-    public Map<String, Object> initiateNegotiation(@RequestBody(description = DSPACE_NEGOTIATION_CONTRACT_REQUEST_MESSAGE, required = true)
+    public Response initiateNegotiation(@RequestBody(description = DSPACE_NEGOTIATION_CONTRACT_REQUEST_MESSAGE, required = true)
                                                    JsonObject jsonObject,
                                                    @HeaderParam(AUTHORIZATION) String token) {
+        ContractNegotiation negotiation;
+        try {
+            negotiation = handleMessage(MessageSpec.Builder.newInstance(ContractRequestMessage.class)
```

Comment:
please check my comments in `DspTransferProcessApiController`, they apply here as well